src/statemachine/statemachine_coroutine.js

```diff
@@ -0,0 +1,225 @@
+pc.extend(pc, (function () {
```

Comment:
We could probably move co-routine into core_coroutine.js. Especially if we remove the dependency on the component system.

This is broader than just state machines.